src/wasm-ir-builder.h

```diff
@@ -369,42 +379,39 @@ class IRBuilder : public UnifiedExpressionVisitor<IRBuilder, Result<>> {
     static ScopeCtx makeIf(If* iff, Name originalLabel, Type inputType) {
       return ScopeCtx(IfScope{iff, originalLabel}, inputType);
     }
-    static ScopeCtx makeElse(If* iff,
-                             Name originalLabel,
-                             Name label,
-                             bool labelUsed,
-                             Type inputType,
-                             Index inputLocal) {
-      return ScopeCtx(
-        ElseScope{iff, originalLabel}, label, labelUsed, inputType, inputLocal);
+    static ScopeCtx makeElse(ScopeCtx&& scope) {
+      scope.scope = ElseScope{scope.getIf(), scope.getOriginalLabel()};
+      scope.resetForDelimiter(/*keepInput=*/true);
+      return scope;
     }
     static ScopeCtx makeLoop(Loop* loop, Type inputType) {
       return ScopeCtx(LoopScope{loop}, inputType);
     }
     static ScopeCtx makeTry(Try* tryy, Name originalLabel, Type inputType) {
       return ScopeCtx(TryScope{tryy, originalLabel}, inputType);
     }
-    static ScopeCtx makeCatch(Try* tryy,
-                              Name originalLabel,
-                              Name label,
-                              bool labelUsed,
-                              Name branchLabel) {
-      return ScopeCtx(
-        CatchScope{tryy, originalLabel}, label, labelUsed, branchLabel);
-    }
-    static ScopeCtx makeCatchAll(Try* tryy,
-                                 Name originalLabel,
-                                 Name label,
-                                 bool labelUsed,
-                                 Name branchLabel) {
-      return ScopeCtx(
-        CatchAllScope{tryy, originalLabel}, label, labelUsed, branchLabel);
+    static ScopeCtx makeCatch(ScopeCtx&& scope, Try* tryy) {
+      scope.scope = CatchScope{tryy, scope.getOriginalLabel()};
+      scope.resetForDelimiter(/*keepInput=*/false);
+      return scope;
+    }
+    static ScopeCtx makeCatchAll(ScopeCtx&& scope, Try* tryy) {
+      scope.scope = CatchAllScope{tryy, scope.getOriginalLabel()};
+      scope.resetForDelimiter(/*keepInput=*/false);
+      return scope;
     }
     static ScopeCtx
     makeTryTable(TryTable* trytable, Name originalLabel, Type inputType) {
       return ScopeCtx(TryTableScope{trytable, originalLabel}, inputType);
     }
-
+    void resetForDelimiter(bool keepInput) {
```

Comment:
Perhaps add a comment for this? Reading e.g. `makeElse` it's not obvious to me what this does, or what "input" refers to in "keepInput".